src/components/data_submission/DataAccessGovernance.js

```diff
@@ -40,22 +94,92 @@ export const DataAccessGovernance = (props) => {
   const addNewConsentGroup = useCallback(() => {
     setConsentGroupsState((consentGroupsState) => {
       const newConsentGroupsState = cloneDeep(consentGroupsState);
-      newConsentGroupsState.push({
+      newConsentGroupsState?.push({
         consentGroup: {},
         key: Math.max(0, ...consentGroupsState.map((state) => state.key)) + 1,
         nihInstitutionalCertificationFile: null,
         editMode: true,
         valid: false,
+        disableFields: false,
       });
 
       return newConsentGroupsState;
     });
   }, []);
 
+  // extract consent groups from datasets
+  const prefillConsentGroups = useCallback(async () => {
+    // from datasets array to consent group array (needs to match the existing consent group structure)
+    var consentGroups = await Promise.all(datasets?.map(async (dataset, idx) => {
+      const dataUse = await normalizeDataUse(dataset?.dataUse);
+      // DAC is required if openAcess is false
+      const openAccess = extract('Open Access', dataset);
+      const dac = openAccess ? undefined : await DAC.get(dataset?.dacId);
+      const dacs = openAccess ? undefined : await DAC.list();
+
+      // extract the consent group data from the dataset
+      return {
+        consentGroup: {
+          consentGroupName: dataset.name,
+
+          // primary
+          generalResearchUse: dataUse.generalUse,
+          hmb: dataUse.hmbResearch,
+          diseaseSpecificUse: dataUse.diseaseRestrictions,
+          poa: dataUse.populationOriginsAncestry,
+          openAccess: openAccess,
+          otherPrimary: dataUse.other,
+
+          // secondary
+          nmds: dataUse.methodsResearch,
+          gso: dataUse.geneticStudiesOnly,
+          pub: dataUse.publicationResults,
+          col: dataUse.collaboratorRequired,
+          irb: dataUse.ethicsApprovalRequired,
+          gs: dataUse.geographicalRestrictions,
+          mor: dataUse.publicationMoratorium,
+          npu: dataUse.commercialUse,
+          otherSecondary: dataUse.secondaryOther,
+          url: extract('URL', dataset),
+          dataLocation: extract('Data Location', dataset),
+          numberOfParticipants: extract('Number Of Participants', dataset),
+          fileTypes: [
+            {
+              fileType: extractFileTypes('File Types', 'fileType', dataset),
+              functionalEquivalence: extractFileTypes('File Types', 'functionalEquivalence', dataset)
+            }
+          ],
+          dataAccessCommitteeId: { ...dac, dacs },
+        },
+        key: `prefilledConsentGroups[${idx}]`,
+        nihInstitutionalCertificationFile: null,
+        editMode: false,
+        valid: false,
+        disableFields: true,
```

Comment:
I think this state is necessary because when `studyEditMode = true`, it is possible that you will be adding a new consent group in which `editMode = true` to edit and `disableFields = false` because the name, dac, etc should not be disabled when adding new but should be disabled when editing a preexisting dataset. Your logic would work if you are only editing the preexisting consent groups

Reply:
Perhaps instead you could have a `isPrefilled` variable with value set to `true` and `false` otherwise, and then set `disabled: isPrefilled` and `editMode: !isPrefilled`, that way you only need to carry one boolean around for each consent group and it's a little more clear why and when the boolean is set.

Reply:
That's a good idea, thanks!

Reply:
The only issue with this logic is that you can't use `isPrefilled` to determine the state of editMode. `editMode` can be true when prefilled or not. With this, I'm not sure that having `isPrefilled` rather than `disableFields` is much more efficient. I think with both `disableFields` and `isPrefilled`  they are the same in controlling the `disabled:` field on the form components, so it is not necessary to have a condition with `studyEditMode` and I will update that now. I will definitely make the change from `disableFields` to `isPrefilled` if you think it is still worthwhile! I think the main difference would be declaring `isPrefilled` as a variable rather than a state on `consentGroupsState`. Please let me know your thoughts!